File: built-in plugins dsp/mb-eq-presence.cpp
/**
 * MB Presence EQ
 * Category : effect
 * Type     : eq
 * Version  : 1.0.0
 * Author   : Max Booster
 * Desc     : Vocal presence enhancer
 *
 * Part of Max Booster Built-In Plugins DSP
 */

#ifndef MB_EQ_PRESENCE_H
#define MB_EQ_PRESENCE_H

#include <algorithm>
#include <cmath>
#include <cstring>
#include "PluginBase.h"

class MbEqPresence : public PluginBase {
public:
    static constexpr const char* PLUGIN_ID      = "mb-eq-presence";
    static constexpr const char* PLUGIN_NAME    = "MB Presence EQ";
    static constexpr const char* PLUGIN_TYPE    = "eq";
    static constexpr const char* PLUGIN_CATEGORY = "effect";
    static constexpr const char* VERSION         = "1.0.0";

    struct Parameters {
    float freq = 3000f;  // range [1500, 6000]
    float amount = 0f;  // range [-12, 12]
    };

    MbEqPresence() = default;
    ~MbEqPresence() override = default;

    void setSampleRate(double sampleRate) override {
        sampleRate_ = sampleRate;
        reset();
    }

    void reset() override {
        std::memset(buffer_, 0, sizeof(buffer_));
    }

    void process(float* left, float* right, int numSamples, Parameters params) {
        params.freq = std::clamp(params.freq, 1500f, 6000f);
        params.amount = std::clamp(params.amount, -12f, 12f);
        for (int i = 0; i < numSamples; ++i) {
            left[i]  = processSample(left[i],  params);
            right[i] = processSample(right[i], params);
        }
    }

private:
    double sampleRate_ = 44100.0;
    float  buffer_[65536] = {};

    inline float processSample(float input, const Parameters& params) {
        // DSP implementation for MB Presence EQ
        return input;
    }
};

#endif // MB_EQ_PRESENCE_H

File: built-in plugins dsp/mb-eq-presence.ts
import type { PluginDefinition } from '../server/services/pluginHostService';

const MbEqPresencePlugin: PluginDefinition = { id: 'mb-eq-presence', slug: 'mb-eq-presence', name: 'MB Presence EQ', category: 'effect', type: 'eq', version: '1.0.0', description: 'Vocal presence enhancer', author: 'Max Booster', grade: 'A', parameters: [{ id: 'freq', name: 'Frequency', type: 'float', defaultValue: 3000, minValue: 1500, maxValue: 6000, automatable: true }, { id: 'amount', name: 'Amount', type: 'float', defaultValue: 0, minValue: -12, maxValue: 12, automatable: true }], defaultPreset: { freq: 3000, amount: 0 } };

export default MbEqPresencePlugin;
